### PR TITLE
Add liboqs integration to CMake build system

### PR DESCRIPTION
**Pull request by darktohka**
# Description
Currently, compiling WolfSSL with liboqs is only possible with the configure system.

I've created this PR to add liboqs support to the CMake build system. Now, CMake can detect liboqs and link to it when building with WOLFSSL_OQS on.

I've added a new CMake option: `WOLFSSL_OQS`, which is disabled by default. When enabled, CMake will look for the liboqs library, and, if found, will link and include it to the wolfssl library. This matches the behavior of the configure system.

# Testing

I've tested this option with multiple configurations:
- WOLFSSL_OQS off, no liboqs installed on system --> nothing happens
- WOLFSSL_OQS on, no liboqs installed on system --> nothing happens
- WOLFSSL_OQS on, liboqs installed on system as shared library --> liboqs is successfully linked to wolfssl dynamically
- WOLFSSL_OQS on, liboqs installed on system as static library --> liboqs is successfully linked to wolfssl statically

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -377,6 +379,24 @@ endif()
 
 set(WOLFSSL_SLOW_MATH "yes")
 
+# liboqs
+add_option(WOLFSSL_OQS
+    "Enable integration with the OQS (Open Quantum Safe) library (default: disabled)"
```
Comment (anhu) [inline CMakeLists.txt:384]: "library" ---> "liboqs library"

**Review by anhu — CHANGES_REQUESTED** [review]

**Comment by darktohka** [comment]
Thank you @anhu! I've e-mailed the address you've given me and will update this thread when I hear back.

**Comment by anhu** [comment]
On-hold until contributor agreement has been approved.

**Review by anhu — APPROVED** [review]

**Comment by embhorn** [comment]
Approved as contributor.

**Comment by dgarske** [comment]
retest this please

**Comment by anhu** [comment]
retest this please

**Comment by anhu** [comment]
Hi @darktohka , 

The failing test indicates that you will need to add an entry into `cmake/include.am` for the new file that you created. 

Warm regards, Anthony

**Comment by darktohka** [comment]
Thank you @anhu, I have made the requested change.

**Comment by anhu** [comment]
Excellent!  Once the tests pass, I will squash and merge your changes.

**Pull request merged**